### PR TITLE
Move `Rack::Logger` from `rack` to `rack-contrib`.

### PR DESCRIPTION
**Pull request by ioquatix**

### CODE REVIEW

**Comment by Earlopain** [comment]
How would this work?

> The middleware doesn't have any dependencies other than rack and the ruby standard library.

It's std now but not for long

**Comment by Earlopain** [comment]
I think, if you're moving this here anyways, does it make sense to require the user to pass in an object that adheres to the rack logger interface themselves

**Comment by ioquatix** [comment]
The current plan, is for us to wait and see if anyone complains about the deprecation. If no one complains, we'll probably remove it with no replacement.

**Comment by mpalmer** [comment]
I'm not sure I understand the rationale for moving this to `rack-contrib`.  My understanding of SPEC is that `rack.logger` *must* be set to some logger-like object.  This means that either:

1. Rack (or servers that implement rack) will provide a default (doesn't-depend-on-logger) implementation that conforms to SPEC; or
2. Every single application in existence will have to either provide their own `rack.logger` implementation or else include `Rack::Logger` as their first middleware, making `rack-contrib` *de facto* a dependency of basically every app.

Since I understand that [(1) is the plan](https://github.com/rack/rack/issues/2190#issuecomment-2152383854), I kinda feel like this middleware would get essentially zero usage.

**Comment by Earlopain** [comment]
`rack.logger` is optional, per my understanding. From `Rack::Lint`: https://github.com/rack/rack/blob/38b178e9ce3b30453cd5340a92093128bfb4fa36/lib/rack/lint.rb#L232

**Comment by mpalmer** [comment]
Ah, thanks for pointing that out.  I'd somehow skipped over that when refreshing my cache of `SPEC`.  That explains why moving this middleware over here makes sense.

File: lib/rack/contrib/logger.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'logger'
```
Comment (mpalmer) [inline lib/rack/contrib/logger.rb:3]: As I think @Earlopain pointed out earlier, the general criteria for inclusion in `rack-contrib` is "no new external deps", and with `Logger` getting evicted with Ruby 3.5, that gets a bit messy.  Are you up for writing a super-slim pseudo-logger that just dumps to `rack.errors`?

Comment (ioquatix) [reply lib/rack/contrib/logger.rb:3]: Yes, it's one idea, I started writing it but then just thought, why am I doing this?

**Comment by ioquatix** [comment]
For now, let's just close this PR.

**Pull request closed without merging**